### DISCUSSION TITLE
Is it possible to show profiling results made on the CLI on the WebGUI?

### DISCUSSION THREAD

**Discussion by 3rc4n** [NoiseByNorthwest/php-spx#155] [Q&A]
I got a simple script with a while(true) loop and added spx_profiler_start() and spx_profiler_stop() within the loop. Between these method calls, there's my echo command to output something.

The CLI command is the following: SPX_ENABLED=1 SPX_REPORT=full SPX_AUTO_START=0 php myTestScript.php

The web GUI itself is working nicely when it comes to track my website calls via webbrowser, which is on the same server where I am using the CLI script. I can click on the entries seen in the results table, I can see the graphs, etc. But I cannot see any entries involving the CLI calls.

Are there any settings I did not consider yet?

**Comment by NoiseByNorthwest** [comment]
It should work.
Are the corresponding report files generated in the data directory ?
Is the data directory used by CLI the same as the one used for HTTP requests ?

**Comment by ozanhazer** [comment]
It's in the documentation, but not very clear at first glance. Also, Laravel Herd does not output the key, which makes it even harder to figure out.

```bash
# Check the spx data dir:
❯ php -i|grep spx
spx.data_dir => /tmp/spx => /tmp/spx
...

# List the files
❯ ls -1 /tmp/spx
spx-full-20251015_181640-Ozans-MacBook-Pro.local-88862-694507980.json
spx-full-20251015_181640-Ozans-MacBook-Pro.local-88862-694507980.txt.gz
spx-full-20251015_182137-Ozans-MacBook-Pro.local-89113-1403640627.json
spx-full-20251015_182137-Ozans-MacBook-Pro.local-89113-1403640627.txt.gz
```

Then use `/report.html` for `SPX_UI_URI` and the file name (without the extension) for the `key` parameter. E.g.:

`http://127.0.0.1/?SPX_UI_URI=/report.html&key=spx-full-20251015_181640-Ozans-MacBook-Pro.local-88862-694507980`
